### DISCUSSION TITLE
Visualize hooks in Kedro-Viz

### DISCUSSION THREAD

**Discussion by galenseilis** [kedro-org/kedro#3880] [Idea]
## Description

As per [this request](https://kedro-org.slack.com/archives/C03RKP2LW64/p1705533826391179?thread_ts=1705525951.357439&cid=C03RKP2LW64) I am opening up this feature request. The general idea is for the hooks to be visualized in some way on Kedro-viz. 

The only design aspect I would suggest is that the symbology on Kedro-Viz should reflect what type of triggers the hook has.

## Context

When I open up a visualization by Kedro-Viz I would like to quickly learn about what things exist, including hooks. In the current state I would have to look into the source code to get an overview of the hooks currently registered for use in the product. The idea I like behind Kedro-Viz is to have a quick overview of any of the information processing specific to that project (i.e. stuff that isn't just native Kedro behaviour). To me getting at least a partial understanding of what hooks exist (and the type of triggers they have) would be helpful for this quick summary of a project.

## Possible Implementation

I don't have any specific recommendations on implementation at this moment.

## Possible Alternatives

- Not everything needs to be visualized with Kedro-Viz

**Comment by datajoely** [comment]
An MVP showing the names of hooks register would be a great first start

Hooks fall into the following categories:

https://docs.kedro.org/en/stable/kedro.framework.hooks.specs.html

At a high level hooks basically fall into 3 categories:

- Stuff that happens during the pipeline creation (not visalisable on the flow chart)
- Stuff that happens during a pipeline run to every part of the pipeline (maybe visalisable on the flow chart)
- Stuff that happens after a pipeline run (not visalisable on the flow chart)

**Comment by NeroOkwa** [comment]
Backlog grooming notes:

> To me getting at least a partial understanding of what hooks exist (and the type of triggers they have) would be helpful for this quick summary of a project.

@galenseilis thanks for this. Can you provide more context on what problem this solves for you.

**Reply by galenseilis** [reply]
I'm not confident what additional context to add. The problem solved is to be able to quickly learn more about what's in a project from peeking in kedro-viz. Is there something you feel is missing?
